backstory/src/main/java/com/google/sps/story/PromptManager.java

```diff
@@ -36,14 +38,18 @@
    * Initialize keywords and randomness parameters.
    *
    * @param keywords A list of Strings containing keywords for prompts.
+   * @param locations A list of Strings containing potential locations for prompts.
    * @throws IllegalArgumentException If input list is null.
    */
-  public PromptManager(List<String> keywords) throws IllegalArgumentException {
+  public PromptManager(List<String> keywords, List<String> locations)
+      throws IllegalArgumentException {
     // Check for null input.
-    if (keywords == null) {
-      throw new IllegalArgumentException("Input list cannot be null.");
+    if (keywords == null || locations == null) {
+      throw new IllegalArgumentException("Input lists cannot be null.");
     }
+
     this.keywords = keywords;
+    this.locations = locations;
```

Comment:
Should probably make a (preferably immutable) copy, so that the values in `location` don't change while this instance is using them.

Reply:
That's a great suggestion. I'll add this.

Reply:
Done